src/main/java/wedlog/address/logic/parser/ParserUtil.java

```diff
@@ -203,4 +203,18 @@ public static <R> R parseOptionally(Optional<String> optionalString, ParserFunct
     }
 
     //@@author
+
+    /**
+     * Returns the result of parsing {@code String} with the given
+     * parser function if {@code string} is not empty, else returns null.
+     */
+
+    public static <R> R parseIfNotBlank(String string, ParserFunction<R> parserFunction)
+            throws ParseException {
+
+        if (!string.isBlank()) {
+            return parserFunction.parse(string);
+        }
+        return null;
```

Comment:
Perhaps we should add a testcase for this method returning null?

Reply:
Good catch! Thank you 😄